nushell_completions.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2013-2022 The Cobra Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cobra
+
+import (
+	"bytes"
+	"io"
+	"os"
+)
+
+func (c *Command) GenNushellCompletion(w io.Writer) error {
```

Comment:
I suggest this function accept a bool parameter to turn off descriptions.  We do that for the other shells.
So, if the program call this function with `false` (GenNushellCompletion(w, false)`) we should generate the same script but instead of using `__complete` we should use `__completeNoDesc`.   This will tell cobra not to include any descriptions with the completions.

Reply:
Thanks for all the feedback, I work through as quickly as I can.

On the directives and adding a space, I'll dig around the nushell source and see if there is anything there. External completers are a pretty new feature, and I _think_ it really only allows you to return a list of completions of a list of strings for values (names) or a list of values and descriptions. I am not sure there is any way to effect behavior yet. I'll look around to be sure.

On descriptions, the description on/off flag will impact all cobra commands. I could make it easy to adjust after the fact. Thoughts?

Reply:
> On descriptions, the description on/off flag will impact all cobra commands. I could make it easy to adjust after the fact. Thoughts?

Good point.
Let's leave this point for later. Your solution may evolve in a way that helps that point. We'll see.

Reply:
The file completion was breaking do to the completer returning an empty array instead of null. I was able to implement ShellCompDirectiveNoFileComp using this.

I wasn't able to implement ShellCompDirectiveFilterFileExt or ShellCompDirectiveFilterDirs. There isn't a way to handle this in nushell via external completers.

My fix for -n= is a little bit hacky. I returned the flag as part of the completion to keep it from replacing the flag. It looks a little odd but it worked:

<img width="783" alt="Screen Shot 2022-11-27 at 11 31 14 AM" src="https://user-images.githubusercontent.com/56345/204160929-050b5876-4715-4005-b5c1-906382277c5a.png">

Reply:
> My fix for -n= is a little bit hacky. I returned the flag as part of the completion to keep it from replacing the flag. It looks a little odd but it worked

This is fine. We do exactly the same for the other shells 👍